core/regex_test.go

```diff
@@ -50,3 +50,37 @@ func TestRegexMatch(t *testing.T) {
 		})
 	}
 }
+
+func TestWildCardMatch(t *testing.T){
+	tests := []struct{
+		pattern string
+		key 	string
+		want	bool
+	}{
+		{"a*b*c", "axybzxc", true},
+		{"", "", true},
+		{"abc", "abc", true},
+		{"a?c", "abc", true},
+		{"ac?", "abc", false},
+		{"ab?", "abc", true},
+		{"?bc", "xbc", true},
+		{"a*c", "abbbbbbc", true},
+		{"abc*", "abcdef", true},
+		{"*abc", "xyzabc", true},
+		{"a*b*c", "axbyczc", true},
+		{"abc", "abd", false},
+		{"abc", "ab", false},
+		{"a*b?c*d", "axbyczdxxd", true},
+		{"*?*", "abc", true},
+		{"a*", "", false},
+		{"", "abc", false},
```

Comment:
We should probably use the same test cases for testing both the new and the old method.

Reply:
Ok